otter/test/utils.py

```diff
@@ -592,6 +594,21 @@ def alist_get(data, key):
     raise KeyError(key)
 
 
+def resolve_effect(effect, result, is_error=False):
+    """
+    Just like :func:`effect.testing.resolve_effect`, except it performs a
+    type-check on ``result`` based on a declared type in the intent's
+    ``intent_result_type`` attribute.
+    """
+    if not is_error:
+        pred = getattr(effect.intent, 'intent_result_pred', None)
```

Comment:
Non-blocking - all the ones without the attribute are the ones that come from `effect`, right?  `ParallelEffect`, `Delay`, etc?  I'm wondering if it's worth checking if the intent are those in particular, to make sure that all of the intents defined in otter all have the predicate?

It may be too much trouble, but even if not, I'm fine with doing that in another PR.